test/testdata/resolver/payload_superclass_redefinition.rb

```diff
@@ -0,0 +1,8 @@
+# typed: true
+
+class Parent
+end
+
+class IRB::RelineInputMethod < Parent
+                             # ^^^^^^ error: Parent of class `IRB::RelineInputMethod` redefined from `IRB::InputMethod` to `Parent`
```

Comment:
Hmm, I'm wondering why this error is reported here, but not in this test https://github.com/sorbet/sorbet/blob/master/test/cli/suppress-payload-superclass-redefinition-for/double.rb?

```ruby
# typed: true

class Parent
end

class Complex < Parent; end
class IRB::RelineInputMethod < Parent
end
```

It seems like the only difference is the amount of redefined constants in a file.

Reply:
That test runs with the flag `suppress-payload-superclass-redefinition-for` https://github.com/sorbet/sorbet/blob/12e21c69b7bb8d2bf75c5110da023c7ab2b7c17e/test/cli/suppress-payload-superclass-redefinition-for/test.sh#L28.